design/pushed-authorization-requests.md

```diff
@@ -0,0 +1,350 @@
+    -   **invalid_client** Client authentication failed, due to missing or invalid client credentials.
+    -   **unauthorized_client** The client is not registered for the requested `response_type`.
+    -   **unsupported_response_type** The server doesn't support the requested `response_type`.
+-   **[ error_description ]** Optional text providing additional information about the error that occurred.
+
+-   **[ error_uri ]** Optional URI for a web page with information about the error that occurred.
+
+
+### Authorization Request
+
+The client uses the request URI value to create the subsequent authorization request by directing the user-agent to make an HTTP request to the authorization server's authorization endpoint like the following:
+
+````
+GET /authorize?client_id=s6BhdRkqt3&request_uri=urn%3Aietf%3Aparams
+       %3Aoauth%3Arequest_uri%3Abwc4JK-ESC0w8acc191e-Y1LTC2 HTTP/1.1
+     Host: as.example.com
+````
+
+## Implementation details
+
+### Server Metadata require_pushed_authorization_requests parameter 
+
+when `false`
+
+there is no mandatory PAR request before autorization endpoint, the client may use it or not
+
+when `true`
+
+PAR request is mandatory prior autorization endpoint and reject any authorization request without a request URI issued from the PAR endpoint, clients must use it
+
+Classes/methods affected:
+
+* org.keycloak.protocol.oidc.OIDCConfigAttributes
+* org.keycloak.protocol.oidc.OIDCWellKnownProvider
+    * getConfig()
+
+### Server Metadata request_uri_lifespan configuration
+The default lifetime of request_uri should be 60 seconds
+
+Classes/methods affected:
+
+* org.keycloak.protocol.oidc.OIDCConfigAttributes
+* org.keycloak.protocol.oidc.OIDCWellKnownProvider
+    * getConfig()
+
+### Client Metadata require_pushed_authorization_requests
+
+when `false`, PAR is not mandatory
+
+when `true`, PAR is mandatory
+
+Classes/methods affected:
+
+* org.keycloak.models.ClientModel
+* org.keycloak.models.cache.infinispan.ClientAdapter
+* org.keycloak.models.cache.infinispan.entities.CachedClient
+* org.keycloak.models.jpa.ClientAdapter
+* org.keycloak.models.jpa.entities.ClientEntity
+* org.keycloak.models.map.client.AbstractClientEntity
+
+### The PAR endpoint 
+
+(1) : Authenticate the client in the same way as at the token endpoint
+
+(2) : Accept The OAuth 2.0 authorization request parameters
+
+(3) : Reject the request if the "request_uri" authorization request parameter is provided.
+
+(4) : Validate the pushed request as it would an authorization request sent to the authorization endpoint
+
+(5) : Generate request_uri
+
+(6) : save the Auth Request+request_uri+request_uri_lifespan
+
+(7) : return PAR Response
+
+Classes/methods to be added:
+
+```java
+/**
+ * OAuth 2.0 PAR request endpoint
+ */
+public class ParEndpoint extends AuthorizationEndpointBase {
```

Comment:
I will change the design accordingly to the implementation, we will reuse some checks from AuthorizationEndpoint in ParValidationService. But yes it will be good to have a common base class for OIDC Authz Endpoint Checks.